File: app/src/main/java/com/example/bindechexclock/ui/notifications/NotificationsFragment.java
package com.example.bindechexclock.ui.notifications;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider; // Changed
import androidx.annotation.Nullable; // Added for onViewCreated
import android.os.Bundle; // Added for onViewCreated

import com.example.bindechexclock.MainActivity;
import com.example.bindechexclock.R;
import com.example.bindechexclock.TimeManager;

public class NotificationsFragment extends Fragment {

    private NotificationsViewModel notificationsViewModel; // Changed to private

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        notificationsViewModel =
                new ViewModelProvider(this).get(NotificationsViewModel.class); // Changed
        View root = inflater.inflate(R.layout.fragment_notifications, container, false);
        final TextView textView = root.findViewById(R.id.text_notifications);
        notificationsViewModel.getText().observe(getViewLifecycleOwner(), new Observer<String>() {
            @Override
            public void onChanged(@Nullable String s) {
                textView.setText(s);
            }
        });
        return root;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        if (getActivity() instanceof MainActivity) {
            MainActivity mainActivity = (MainActivity) getActivity();
            TimeManager timeManager = mainActivity.getTimeManager();

            timeManager.getHexTimeLiveData().observe(getViewLifecycleOwner(), new Observer<String>() {
                @Override
                public void onChanged(String hexTimeValue) {
                    // The old TimeManager used "  ->  " for hex, let's replicate that.
                    // String prefix = getString(R.string.time_in) + " " + getString(R.string.title_hex) + "\n";
                    String prefix = getString(R.string.time_in) + " " + getString(R.string.title_hex) + "  ->  ";
                    notificationsViewModel.setmText(prefix + hexTimeValue); // hexTimeValue likely includes \n
                }
            });
        }
    }
}

File: app/src/main/java/com/example/bindechexclock/ui/notifications/NotificationsFragment.kt
package com.example.bindechexclock.ui.notifications

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.viewModels
import com.example.bindechexclock.MainActivity
import com.example.bindechexclock.R
import com.example.bindechexclock.TimeManager
import com.example.bindechexclock.databinding.FragmentNotificationsBinding // Import view binding class

class NotificationsFragment : Fragment() {

    private val notificationsViewModel: NotificationsViewModel by viewModels()
    private var _binding: FragmentNotificationsBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentNotificationsBinding.inflate(inflater, container, false)

        notificationsViewModel.text.observe(viewLifecycleOwner) { s ->
            binding.textNotifications.text = s
        }
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        (activity as? MainActivity)?.let { mainActivity ->
            val timeManager: TimeManager = mainActivity.timeManager

            timeManager.hexTimeLiveData.observe(viewLifecycleOwner) { hexTimeValue ->
                val prefix = getString(R.string.time_in) + " " + getString(R.string.title_hex) + "  ->  "
                // Use updateText from the Kotlin ViewModel
                notificationsViewModel.updateText(prefix + hexTimeValue)
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
